src/Bounds.cpp

```diff
@@ -1340,6 +1340,34 @@ class Bounds : public IRVisitor {
         interval = result;
     }
 
+    void visit(const VectorReduce *op) override {
+        TRACK_BOUNDS_INTERVAL;
+        op->value.accept(this);
+        int factor = op->value.type().lanes() / op->type.lanes();
+        switch (op->op) {
+        case VectorReduce::Add:
+            if (interval.has_upper_bound()) {
+                interval.max *= factor;
+            }
+            if (interval.has_lower_bound()) {
+                interval.min *= factor;
+            }
+            break;
+        case VectorReduce::Mul:
+            interval = Interval::everything();
```

Comment:
Nit: aren't there *some* cases we could exploit here? e.g., what if the type is float and we know the bounds are [0.0, 1.0]?